andela/vof-deployment-scripts
# Ft target pool 154358789

#### What does this PR do?

This P.R adds the feature target pool on google cloud platform. The purpose is to ensure that all connections from a client will end up on the same instance as long as they use the same protocol and the instance stays healthy.

The P.R also adds information on how to create a vof base image from the branch on enhancing documentation.

#### Description of Task to be completed?

Currently the vof-tracker spins instances horizontally, the requests using the load balancer algorithm will serve requests in a round robin manner to a new instance for each request. However this conflicts with the csrf architecture for the ruby application where after a get request to an instance the instance will get a csrf token specific to the user, however any subsequent PUT POST or DELETE request will be served to other instances on the instance group manager that do not contain the csrf token generated.

This P.R implements load balancer stickiness ensuring that client requests will end up on the same instance.

#### How should this be manually tested?

Step 1. Access the vof url, login using a test email and try to submit forms. The error to be raised should be a 422 Error. 

Step 2. Deploy the application using this branch on circle CI, or rebuild the infrastructure using terraform refer to [Environment Creation](docs/environment.md) .

Step 3.  Ensure that instances are added to the target pool.

To ensure that your virtual machines are added on the target pool configure your gcloud project to the project where you are hosting the vof app buy running the command `gcloud config set project <project-name>`. Ensure that your gcloud account has the rights to access the specific project. Run the command `gcloud compute target-pools describe target pool` to get a list of instances that have been added to the target pool.

Step 4. Repeat step 1. The error 422 should not be raised.


#### What are the relevant pivotal tracker stories?

https://www.pivotaltracker.com/story/show/154358789
